src/test-jdk14/java/com/fasterxml/jackson/databind/failing/RecordDeserialization3906Test.java

```diff
@@ -0,0 +1,159 @@
+
+        assertEquals(new Record3906(null, 0), recordDeser);
+    }
+
+    /*
+    /**********************************************************
+    /* Passing Tests : Suggested work-arounds
+    /*                                for future modifications
+    /**********************************************************
+     */
+
+    public void testEmptyJsonToRecordWorkAround() throws JsonProcessingException {
+        ObjectMapper mapper = newJsonMapper();
+        mapper.setVisibility(PropertyAccessor.ALL, Visibility.NONE);
+        mapper.setVisibility(PropertyAccessor.CREATOR, Visibility.ANY);
+
+        Record3906 recordDeser = mapper.readValue("{}", Record3906.class);
+
+        assertEquals(new Record3906(null, 0), recordDeser);
+    }
+
+    public void testEmptyJsonToRecordCreatorsVisibile() throws JsonProcessingException {
+        ObjectMapper mapper = newJsonMapper();
+        mapper.setVisibility(PropertyAccessor.CREATOR, Visibility.NON_PRIVATE);
+
+        Record3906 recordDeser = mapper.readValue("{}", Record3906.class);
+        assertEquals(new Record3906(null, 0), recordDeser);
+    }
+
+    public void testEmptyJsonToRecordUsingModule() throws JsonProcessingException {
+        ObjectMapper mapper = newJsonMapper().registerModule(new SimpleModule() {
+            @Override
+            public void setupModule(SetupContext context) {
+                super.setupModule(context);
+                context.insertAnnotationIntrospector(new NopAnnotationIntrospector() {
+                    @Override
+                    public VisibilityChecker<?> findAutoDetectVisibility(AnnotatedClass ac,
+                                                                         VisibilityChecker<?> checker) {
+                        return ac.getType().isRecordType()
+                                ? checker.withCreatorVisibility(JsonAutoDetect.Visibility.NON_PRIVATE)
+                                : checker;
+                    }
+                });
+            }
+        });
+
+        Record3906 recordDeser = mapper.readValue("{}", Record3906.class);
+        assertEquals(new Record3906(null, 0), recordDeser);
+    }
+
+    public void testEmptyJsonToRecordDirectAutoDetectConfig() throws JsonProcessingException {
+        ObjectMapper mapper = newJsonMapper();
+
+        Record3906Annotated recordDeser = mapper.readValue("{}", Record3906Annotated.class);
+        assertEquals(new Record3906Annotated(null, 0), recordDeser);
+    }
+
+    public void testEmptyJsonToRecordJsonCreator() throws JsonProcessingException {
+        ObjectMapper mapper = newJsonMapper();
+
+        Record3906Creator recordDeser = mapper.readValue("{}", Record3906Creator.class);
+        assertEquals(new Record3906Creator(null, 0), recordDeser);
+    }
+
+    public void testEmptyJsonToRecordUsingModuleOther() throws JsonProcessingException {
+        ObjectMapper mapper = newJsonMapper().registerModule(new SimpleModule() {
```

Comment:
Could you change this to use `mapperBuilder()` instead? That way it'll merge cleanly into master.

Reply:
Right, that one :)